library/Frontend/src/Juvix/Frontend/Types/Base.hs

```diff
@@ -148,6 +150,29 @@ data NameType = NameType'
   }
   deriving (Show, Read, Eq)
 
+--------------------------------------------------------------------------------
+-- Effect Handlers
+--------------------------------------------------------------------------------
+
+data Effect = Eff
+  { effName :: Symbol,
+    effOps :: [Signature],
+    effRet :: Signature
+  }
+  deriving (Show, Read, Eq)
+
+data Operation = Op (FunctionLike Expression)
+  deriving (Show, Read, Eq)
+
+--  A 'Handler', as implemented here, is a set of functions that implement
+-- (at least) one `Effect` interface.
+-- However, Handlers are NOT scoped, meaning that they can't be defined
+-- defined within another function. We CAN fix that, but it requires
+-- us to make some choices, it's wise to have Witch up and running first.
```

Comment:
Handlers not being scoped might actually be a good thing? Having each and every function use it's own handler definition for the same effect could be very confusing. On the other hand it would provide more flexibility.

Also note the double `defined defined` in 169-170.